test/functional/collection_tests.js

```diff
@@ -12,1648 +11,1040 @@ describe('Collection', function() {
-
-    // The actual test we wish to run
-    test: function(done) {
-      var self = this;
-      var client = self.configuration.newClient(self.configuration.writeConcernMax(), {
-        poolSize: 1
+    /**
+     * @ignore
+     */
+    it('should perform collection remove with no callback', function(done) {
+      db.collection('remove_with_no_callback_bug_test', (err, collection) => {
+        expect(err).to.not.exist;
+        collection.insertOne({ a: 1 }, this.configuration.writeConcernMax(), err => {
+          expect(err).to.not.exist;
+          collection.insertOne({ b: 1 }, this.configuration.writeConcernMax(), err => {
+            expect(err).to.not.exist;
+            collection.insertOne({ c: 1 }, this.configuration.writeConcernMax(), err => {
+              expect(err).to.not.exist;
+              collection.remove({ a: 1 }, this.configuration.writeConcernMax(), err => {
+                expect(err).to.not.exist;
+                // Let's perform a count
+                collection.countDocuments((err, count) => {
+                  expect(err).to.not.exist;
+                  expect(count).to.equal(2);
+                  done();
+                });
+              });
+            });
+          });
+        });
       });
+    });
 
-      client.connect(function(err, client) {
-        var db = client.db(self.configuration.db);
-
-        db.collection('shouldCorrectlyHandle0asIdForSave').save({ _id: 0 }, function(err) {
-          test.equal(null, err);
+    /**
+     * @ignore
+     */
+    it('should correctly read back document with null', function(done) {
+      db.createCollection('shouldCorrectlyReadBackDocumentWithNull', {}, (err, collection) => {
+        // Insert a document with a date
+        collection.insertOne({ test: null }, this.configuration.writeConcernMax(), err => {
+          expect(err).to.not.exist;
 
-          db.collection('shouldCorrectlyHandle0asIdForSave').save({ _id: 0 }, function(err) {
-            test.equal(null, err);
-            client.close();
+          collection.findOne((err, result) => {
+            expect(err).to.not.exist;
+            expect(result.test).to.not.exist;
             done();
           });
         });
       });
-    }
-  });
-
-  /**
-   * @ignore
-   */
-  it('should correctly execute update with . field in selector', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
-
-    // The actual test we wish to run
-    test: function(done) {
-      var self = this;
-      var client = self.configuration.newClient(self.configuration.writeConcernMax(), {
-        poolSize: 1
-      });
+    });
 
-      client.connect(function(err, client) {
-        var db = client.db(self.configuration.db);
+    /**
+     * @ignore
+     */
+    it('should throw error due to illegal update', function(done) {
+      db.createCollection('shouldThrowErrorDueToIllegalUpdate', {}, (err, coll) => {
+        try {
```

Comment:
These should be in the form of `expect(() => coll.update({}, null, function() {})).to.throw('document must be a valid JavaScript object');` If they don't pass after that change, then there's something wrong with this test and it was never checking what we thought it should. We should correct it so that it throws an error if an invalid `update` is performed.